src/azul/service/repository_service.py

```diff
@@ -70,12 +73,60 @@ def __init__(self, entity_type: str, entity_id: str):
         super().__init__(f"Can't find an entity in {entity_type} with an uuid, {entity_id}.")
 
 
+@attr.s(frozen=True, auto_attribs=True, kw_only=True)
 class SearchResponseStage(_ElasticsearchStage[ResponseTriple, MutableJSON],
                           metaclass=ABCMeta):
+    file_url_func: FileUrlFunc
+    filters: Filters
 
     def prepare_request(self, request: Search) -> Search:
         return request
 
+    @cached_property
+    def repository_plugin(self) -> RepositoryPlugin:
+        return RepositoryPlugin.load(self.catalog).create(self.catalog)
```

Comment:
We should try to reuse an existing plugin instance. My patch here doesn't quite work because `self.service` is annotated as a being an instance of `DocumentService`, which doesn't have a repository plugin instance, but with a little more work it should be feasible. The easiest fix is to pull `RepositoryService.repository_plugin` up into `DocumentService`, but there might be other options.

```suggestion
    @property
    def repository_plugin(self) -> RepositoryPlugin:
        return self.service.repository_plugin(self.catalog)
```

Reply:
When the subclasses of `SearchResponseStage` are instantiated outside of test code, I believe `self.service` is always an instance of `RepositoryService`, which _does_ have the `repository_plugin` attribute. Hence why I feel this is relatively low-hanging fruit.